lib/vue-sfc-compiler.js

```diff
@@ -0,0 +1,184 @@
+      if(err) {
+        // clear cache for retry
+        delete cache[task.filename];
+        return callback(err);
+      }
+      task.bundle = bundle;
+      callback();
+    });
+  }
+
+  logger.debug(`Compiling Vue SFC: ${task.componentPath}`);
+  async.auto({
+    compile: callback => compile(task, callback),
+    emptyDir: callback => fs.emptyDir(
+      path.dirname(task.cachedFilename), callback),
+    write: ['compile', (results, callback) => {
+      task.bundle = results.compile.bundle;
+      fs.writeFile(task.cachedFilename, task.bundle, 'utf8', callback);
+    }],
+    finish: ['write', (results, callback) => {
+      entry.mtime = task.mtime;
+      entry.cachedFilename = task.cachedFilename;
+      entry.dependencies = results.compile.dependencies;
+      callback();
+    }]
+  }, callback);
+}
+
+function compile({filename, pkg, externals}, callback) {
+  const compiler = webpack({
+    context: '/tmp',
+    mode: 'development',
+    target: 'web',
+    entry: filename,
+    output: {
+      path: path.resolve(__dirname),
+      filename: 'bundle.js',
+      libraryTarget: 'umd'
+    },
+    /*
+      Prevent bundling of certain imported packages and instead retrieve
+      these external dependencies at runtime.
+     */
+    externals,
+    module: {
+      rules: [{
+        test: /\.vue$/,
+        use: {
+          loader: require.resolve('vue-loader'),
+          options: {
+            hotReload: false,
+            cssSourceMap: false,
+            loaders: {
+              // disable babel, unnecessary overhead
+              // TODO: to enable, must do:
+              // npm install babel-loader and babel-core
+              js: '', //require.resolve('babel-loader'),
+              css: require.resolve('vue-style-loader') + '!' +
+                require.resolve('css-loader')
+            }
+          }
+        }
+      }]
+    }
+  });
+
+  compiler.outputFileSystem = new memoryfs();
+
+  compiler.run((err, stats) => {
+    if(err) {
+      return callback(err);
+    }
+    if(stats.compilation.errors.length > 0) {
+      logger.error(`SFC compilation error:\nFile: ${filename}\n` +
+        stats.compilation.errors);
+      return callback(new BedrockError(
+        'Could not compile Vue Single Component File.',
+        'DataError', {
+          'public': true,
+          httpStatusCode: 500,
+          errors: stats.compilation.errors.map(e => String(e))
+        }));
+    }
+
+    const dependencies = [filename, ...stats.compilation.fileDependencies]
+      .filter(dep => typeof dep === 'string');
+    const bundlePath = path.resolve(__dirname, 'bundle.js');
+    const bundle = compiler.outputFileSystem.readFileSync(
+      bundlePath, 'utf8');
+    callback(null, {bundle, dependencies});
+  });
+}
+
+function mostRecentMtime(filename, deps, callback) {
+  let mtime = 0;
+  async.each(deps, (dep, callback) => {
+    fs.stat(dep, (err, stat) => {
+      if(err) {
+        if(filename === dep) {
```

Comment:
is `err` when `filename !== dep` being handled properly?  If so, a comment would be helpful.